homeassistant/util/network.py

```diff
@@ -20,3 +22,14 @@ def is_local(address: Union[IPv4Address, IPv6Address]) -> bool:
     """Check if an address is local."""
     return address in LOCAL_IPS or \
         any(address in network for network in LOCAL_NETWORKS)
+
+
+def get_mac_address(ip_address: str) -> Optional[str]:  # noqa: E501 pylint: disable=redefined-outer-name
+    """Get MAC address for an IP address."""
+    proc = subprocess.run(("arp", "-n", ip_address), stdout=subprocess.PIPE)
```

Comment:
Is there a way that we can do this without relying on an external tool? There must be a PyPI extension that does a similar thing ?

Reply:
https://pypi.org/project/getmac/

This is an active project. I use it in pymysensors.

Reply:
Alternative approach in #24628 uses that